### ISSUE TITLE
How to create Gutenberg block settings?

### ISSUE THREAD

**Issue by speedwheel** [htmlburger/carbon-fields#649]
Hello,

Is there a way to create Gutenberg block settings?
I've attached an image to understand better what I am talking about:

![block-settings](https://user-images.githubusercontent.com/5404341/51430917-57473780-1c2a-11e9-9d54-4b70e563a8c3.JPG)

**Comment by emohamed** [comment]
Hi @speedwheel 

Just to make sure that we're on the same page: you're looking for API to define additional settings for built-in and 3rd party blocks, rather than setup settings for blocks created through Carbon Fields? 

If that's the case -- currently this is not supported by Carbon Fields, and it's not currently on our roadmap. 

It's an interesting idea, though. I'm leaving the issue open for time being, we might revisit it in future. 

Thanks!

**Comment by speedwheel** [comment]
@emohamed Yes that's correct (additional settings), ok thank you for your answer.

**Comment by brianpurkiss** [comment]
What about creating settings for Blocks created through Carbon Fields?

**Comment by SimJoSt** [comment]
@emohamed I was wondering about this as well. I couldn't find any details in the docs regarding the sidebar other than making the description of the block visible there.  
Is it already possible to add block settings to blocks created with Carbon Fields? If yes, how?
